packages/core/upload/admin/src/pages/App/MediaLibrary/index.js

```diff
@@ -361,24 +365,33 @@ export const MediaLibrary = () => {
 
               {assetCount > 0 && (
                 <>
-                  <AssetList
-                    assets={assets}
-                    onEditAsset={setAssetToEdit}
-                    onSelectAsset={selectOne}
-                    selectedAssets={selected.filter(({ type }) => type === 'asset')}
-                    title={
-                      ((!isFiltering || (isFiltering && folderCount > 0)) &&
-                        assetsData?.pagination?.page === 1 &&
-                        formatMessage(
-                          {
-                            id: getTrad('list.assets.title'),
-                            defaultMessage: 'Assets ({count})',
-                          },
-                          { count: assetCount }
-                        )) ||
-                      ''
-                    }
-                  />
+                  {isGridView ? (
+                    <AssetGridList
+                      assets={assets}
+                      onEditAsset={setAssetToEdit}
+                      onSelectAsset={selectOne}
+                      selectedAssets={selected.filter(({ type }) => type === 'asset')}
+                      title={
+                        ((!isFiltering || (isFiltering && folderCount > 0)) &&
```

Comment:
Sorry, I find this too confusing to understand what's going on, i'd lift it out and some code comment to explain the conditions 👍🏼